src/BlazorWebView/src/Maui/RootComponent.cs

```diff
@@ -14,6 +14,19 @@ namespace Microsoft.AspNetCore.Components.WebView.Maui
 	/// </summary>
 	public class RootComponent
 	{
+		/// <summary>
+		/// Constructs an instance of <see cref="RootComponent"/>.
+		/// </summary>
+		/// <param name="selector">The CSS selector string that specifies where in the document the component should be placed. This must be unique among the root components within the <see cref="BlazorWebView"/>.</param>
+		/// <param name="componentType">The type of the root component. This type must implement <see cref="IComponent"/>.</param>
+		/// <param name="parameters">An optional dictionary of parameters to pass to the root component.</param>
+		public RootComponent(string selector, Type componentType, IDictionary<string, object?>? parameters)
+		{
+			Selector = selector;
+			ComponentType = componentType;
+			Parameters = parameters;
+		}
```

Comment:
I wonder why all the properties on this type are publicly writable, and why they allow null even for `ComponentType`, and why the parameters are on an `IDictionary` rather than `IReadOnlyDictionary`. Perhaps the type just hasn't been through any API review yet but it seems very slack.

@javiercn Do you think now with this PR would be a good time to tighten it up? @Eilon do you know if there was a specific plan to API-review everything in this repo?

Reply:
I would imagine we would do a `single` pass API review over all APIs. I would only expect parameters to be nullable, for example.

Reply:
We did do an API review a while back, but I think maybe these APIs changed? Or we didn't realize the inconsistency? Either way, yes, we should do another one soon-ish.